spec/nmt.md

```diff
@@ -0,0 +1,303 @@
+      NsH()               NsH()              NsH()               NsH()
+        |                   |                 |                   |
+00 6c6561665f30      00 6c6561665f31    01 6c6561665f32      03 6c6561665f33  Namespaced Data Items 
+
+        0                   1                  2                  3           Data Items/Leaves Indices
+``` 
+Figure 1.
+
+## Namespace Proof
+
+NMT supports standard Merkle tree functionalities, including inclusion and range proofs, and offers namespace ID querying and proof generation. 
+The following enumerated cases explain potential outcomes when querying an NMT for a namespace ID.
+
+### Namespace Inclusion Proof
+
+When the queried namespace ID `nID` has corresponding items in the tree with root `T`, the query is resolved by a 
+namespace inclusion proof which consists of:
+1) The starting index `start` and the ending index `end` of the leaves that match `nID`.
+2) Nodes of the tree that are necessary for the regular Merkle range proof of `[start, end)` to `T`. 
+   In specific, the nodes include 1) the [namespaced hash](#namespaced-hash) of the left siblings for the Merkle 
+   inclusion proof of the `start` leaf and 2) the [namespaced hash](#namespaced-hash) of the right siblings of the Merkle inclusion proof of  the `end` leaf. 
+   Nodes are sorted according to the in-order traversal of the tree.
+
+For example, the NMT proof of `nID = 0` in Figure 1 would be `[start = 0, end = 2)` and the Merkle inclusion proof embodies one single tree node i.e., `01 03 52c7c03`.
+
+### Namespace Absence Proof
+
+If the namespace ID being queried falls within the range of namespace IDs in the tree root, but there is no corresponding message, an absence proof will be generated. 
+An absence proof asserts that no message in the tree matches the queried namespace ID `nID`.
+
+The absence proof consists of:
+1) The index of a leaf of the tree that 
+   1) its namespace ID is the smallest namespace ID larger than `nID` and
+   2) the namespace ID of the leaf to the left of it  is smaller than `nID` 
+   3) the namespace ID of the leaf to the right of it is larger than  `nID`.
+2) A regular Merkle inclusion proof for the said leaf to the tree root `T`.
+
+Note that the proof only requires the hash of the leaf, not its underlying message. 
+This is because the aim of the proof is to demonstrate the absence of `nID`.
+
+In Figure 1, if we query for `nID = 2`, we will receive an absence proof since there is no matching item for it. 
+The index of the leaf included in the proof will be `3`, which corresponds to the node `03 03 b4a2792`. 
+The Merkle inclusion proof for this leaf consists of the following nodes, in the order they appear in an in-order traversal of the tree: `00 00 ead8d25` and `01 01 71ca46a`.
+
+
+### Namespace Empty Proof
+If the requested namespace ID falls outside the namespace range represented by the tree root `T`, the query will be resolved with an empty namespace proof.
```

Comment:
Good question, yes, this is actually the case in the current implementation. 
I am not quite sure about its use-case though. In fact, I originally suggested to provide an exclusion proof for such queries instead of returning an empty proof. Here is a link to this conversation in another PR https://github.com/celestiaorg/nmt/pull/94/files#r1098349894. Once the final decision is made, I'll update the spec.

Reply:
Update: To be more exact, my suggestion is part of the following comment in that said PR, please see https://github.com/celestiaorg/nmt/blob/43a8d1e6271aff69cf2bf1cb861f5ff79c0984f9/nmt.go#L180

Reply:
thanks for the link! I think we can resolve this convo because you're already tracking the question in https://github.com/celestiaorg/nmt/pull/94/files#r1098349894